neo/Network/P2P/TaskManager.cs

```diff
@@ -29,28 +28,16 @@ private class Timer { }
         private readonly Dictionary<IActorRef, TaskSession> sessions = new Dictionary<IActorRef, TaskSession>();
         private readonly ICancelable timer = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(TimerInterval, TimerInterval, Context.Self, new Timer(), ActorRefs.NoSender);
 
-        private readonly UInt256 HeaderTaskHash = UInt256.Zero;
-        private bool HasHeaderTask => globalTasks.ContainsKey(HeaderTaskHash);
-
         public TaskManager(NeoSystem system)
         {
             this.system = system;
         }
 
-        private void OnHeaderTaskCompleted()
-        {
-            if (!sessions.TryGetValue(Sender, out TaskSession session))
-                return;
-            session.Tasks.Remove(HeaderTaskHash);
-            DecrementGlobalTask(HeaderTaskHash);
-            RequestTasks(session);
-        }
-
         private void OnNewTasks(InvPayload payload)
         {
             if (!sessions.TryGetValue(Sender, out TaskSession session))
                 return;
-            if (payload.Type == InventoryType.TX && Blockchain.Singleton.Height < Blockchain.Singleton.HeaderHeight)
+            if (payload.Type == InventoryType.TX && Blockchain.Singleton.Height < session.StartHeight)
```

Comment:
I am not sure about this...
StartHeight is not often updated, what are the benefits to use this as a basins? It would make us avoid to request blocks to someone that could potentially help.

On the other hand, update startheight could be costly as well,